src/databricks/labs/ucx/workspace_access/tacl.py

```diff
@@ -86,15 +88,47 @@ def get_apply_task(self, item: Permissions, migration_state: MigrationState):
             return None
         target_grant = dataclasses.replace(grant, principal=target_principal)
         # this has to be executed on tacl cluster, otherwise - use SQLExecutionAPI backend & Warehouse
-        return partial(self._apply_grant, target_grant)
+        object_type, object_id = target_grant.this_type_and_key()
+        return partial(self._apply_grant, object_type, object_id, target_grant)
 
-    def _apply_grant(self, grant: Grant):
+    def _apply_grant(self, object_type: str, object_id: str, grant: Grant):
```

Comment:
You can get object type and ID from grant itself, don't add extras arguments here